### ISSUE TITLE
Fine Tuning Error with Flatten

### ISSUE THREAD

**Issue by jayanthkoushik** [fchollet/keras#2885]
I'm trying to fine tune VGGNet using the code and weights from here: [https://gist.github.com/baraldilorenzo/07d7802847aaad0a35d3](https://gist.github.com/baraldilorenzo/07d7802847aaad0a35d3). I do this by popping layers and adding my own.

```
model.layers.pop()
model.add(Dense(100))
```

This works fine for changing layers after the 'Flatten' layer. But trying to remove the Flatten layer and add it again fails.

```
>>> for _ in xrange(6):
>>>     model.layers.pop()
>>> model.add(Flatten())

...
/home/scratch/jayanthkoushik/transfer-nets/env/lib/python2.7/site-packages/keras/engine/topology.pyc in assert_input_compatibility(self, input)
    373                                         self.name + ': expected ndim >= ' +
    374                                         str(ndim) + ', found ndim=' +
--> 375                                         str(K.ndim(x)))
    376                 else:
    377                     if K.ndim(x) != spec.ndim:

Exception: Input 0 is incompatible with layer flatten_39: expected ndim >= 3, found ndim=2
```

I'm not sure what's special about Flatten that would cause this to happen. I'm using the latest build of both Theano and Keras.

**Comment by joelthchao** [comment]
This is a known issue, don't trust manual pop on `model.layers`. You also need to maintain `model.output`. Use my [code](https://github.com/fchollet/keras/issues/2640#issuecomment-217346104) to achieve it.

**Comment by jayanthkoushik** [comment]
Got it. Thanks!

**Comment by alyato** [comment]
@jayanthkoushik 
I wanna use this pretrained vgg19.  

>  x = Flatten(name='flatten')(x)
        x = Dense(4096, activation='relu', name='fc1')(x)
        x = Dense(4096, activation='relu', name='fc2')(x)
        x = Dense(1000, activation='softmax', name='predictions')(x)
 model = Model(img_input, x)
    return model

 i have only 8 labels to classlify. 
 But it is used by the keras function API ,not the `Sequential` model.
 how can i use the `model.layers.pop()`
like this

>  x = Dense(1000, activation='softmax', name='predictions')(x)
model = Model(img_input, x)
model.layers.pop()
 x = Dense(8, activation='softmax', name='predictions')(x)
 loading weight
return model

Do you give me some advices? Thanks
